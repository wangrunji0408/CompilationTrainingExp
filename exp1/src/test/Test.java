package test;

public class Test {
    public static void main (String [] args) {
	int a, b, c;
	a = 3;
	if (a < 2) {
	    b = 3;
	    c = 5;
	} else {
	    b = 3;
	    c = 6;
	}
	a = -a;
	c = a+b;
	System.out.println(a+" "+b+" "+c);
    }
}
