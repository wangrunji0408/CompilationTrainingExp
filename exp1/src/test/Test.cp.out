Now analyzing test.Test
Method: main
Initialization completed.
entry: {R0=NAC, R1=undef, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
0 in:  {R0=undef, R1=undef, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
0 out: {R0=undef, R1=undef, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
1 in:  {R0=NAC, R1=undef, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
1 out: {R0=NAC, R1=3, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
2 in:  {R0=NAC, R1=3, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
2 out: {R0=NAC, R1=3, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
3 in:  {R0=NAC, R1=3, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
3 out: {R0=NAC, R1=3, R2=3, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
4 in:  {R0=NAC, R1=3, R2=3, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
4 out: {R0=NAC, R1=3, R2=3, R3=6, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
5 in:  {R0=NAC, R1=3, R2=undef, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
5 out: {R0=NAC, R1=3, R2=3, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
6 in:  {R0=NAC, R1=3, R2=3, R3=undef, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
6 out: {R0=NAC, R1=3, R2=3, R3=5, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
7 in:  {R0=NAC, R1=3, R2=3, R3=5, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
7 out: {R0=NAC, R1=3, R2=3, R3=5, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
8 in:  {R0=NAC, R1=3, R2=3, R3=NAC, T-1=undef, T4=undef, T5=undef, T6=undef, T7=undef, T8=undef}
8 out: {R0=NAC, R1=3, R2=3, R3=NAC, T-1=undef, T4=-3, T5=undef, T6=undef, T7=undef, T8=undef}
9 in:  {R0=NAC, R1=3, R2=3, R3=NAC, T-1=undef, T4=-3, T5=undef, T6=undef, T7=undef, T8=undef}
9 out: {R0=NAC, R1=-3, R2=3, R3=NAC, T-1=undef, T4=-3, T5=undef, T6=undef, T7=undef, T8=undef}
10 in:  {R0=NAC, R1=-3, R2=3, R3=NAC, T-1=undef, T4=-3, T5=undef, T6=undef, T7=undef, T8=undef}
10 out: {R0=NAC, R1=-3, R2=3, R3=NAC, T-1=undef, T4=0, T5=undef, T6=undef, T7=undef, T8=undef}
11 in:  {R0=NAC, R1=-3, R2=3, R3=NAC, T-1=undef, T4=0, T5=undef, T6=undef, T7=undef, T8=undef}
11 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=undef, T6=undef, T7=undef, T8=undef}
12 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=undef, T6=undef, T7=undef, T8=undef}
12 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=undef, T7=undef, T8=undef}
13 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=undef, T7=undef, T8=undef}
13 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=undef, T8=undef}
14 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=undef, T8=undef}
14 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
15 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
15 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
16 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
16 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
17 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
17 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
18 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
18 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
19 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
19 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
20 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
20 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
21 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
21 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
22 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
22 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
23 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=undef}
23 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
24 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
24 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
25 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
25 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
26 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
26 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
27 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
27 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
28 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
28 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
29 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
29 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
30 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
30 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
31 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
31 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
32 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
32 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
33 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
33 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
34 in:  {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
34 out: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
exit: {R0=NAC, R1=-3, R2=3, R3=0, T-1=undef, T4=0, T5=NAC, T6=NAC, T7=NAC, T8=3}
Method: <init>
Initialization completed.
entry: {R0=NAC, T-1=undef}
0 in:  {R0=undef, T-1=undef}
0 out: {R0=undef, T-1=undef}
1 in:  {R0=NAC, T-1=undef}
1 out: {R0=NAC, T-1=undef}
2 in:  {R0=NAC, T-1=undef}
2 out: {R0=NAC, T-1=undef}
3 in:  {R0=NAC, T-1=undef}
3 out: {R0=NAC, T-1=undef}
exit: {R0=NAC, T-1=undef}
